src/main/java/org/mitre/dsmiley/httpproxy/ProxyServlet.java

```diff
@@ -223,16 +233,18 @@ protected void initTarget() throws ServletException {
    * HttpClient offers many opportunities for customization.
    * In any case, it should be thread-safe.
    */
-  protected HttpClient createHttpClient(final RequestConfig requestConfig) {
-    HttpClientBuilder clientBuilder = HttpClientBuilder.create().setDefaultRequestConfig(requestConfig);
+  protected HttpClient createHttpClient(final RequestConfig requestConfig, final SocketConfig socketConfig) {
```

Comment:
Lets have no arguments to this method.  The possible range of arguments could expand arbitrarily so lets go the other way and have zero args.  The default impl of this method below can call buildRequestConfig() and buildSocketConfig().

Reply:
Sorry I missed your comment :( -> DONE